### ISSUE TITLE
Merge into Master + Deploy to Production

### ISSUE THREAD

**Issue by cmfcmf** [AdventureLookup/AdventureLookup#89]
Now that we restructured the database and got some other features implemented ~~(only real bug is left in #88)~~, it is time to merge the changes into master and deploy them.

The deployment is a little more complicated, because of the many changes to the data model. It might cause issues we don't expect. When I last tested the upgrade, Adventure Lookup had way less than 200 adventures. Today it has more than 248. The earlier we upgrade, the less likely we run into problems.

@JohnnyFlash do you think you could temporarily shutdown the site, create another database dump for us to test, and if it works, execute the upgrade and re-enable the site? @harkenn what do you think?

Here's the list of commands to execute the upgrade **once we merged all changes into master**.
```bash
# Make sure to checkout master!
git pull
export SYMFONY_ENV=prod 
composer install --no-dev --optimize-autoloader
npm run prod
php bin/console doctrine:migrations:migrate
php bin/console app:migrate-data-model # Not normally part of the upgrade, only this time, will display a bunch of 'WARN: ...' messages and take a while
php bin/console app:elasticsearch:reindex
rm -R var/sessions/prod/*
chown www-data:www-data -R . # Only required if update is executed as root
```

**Comment by JohnnyFlash** [comment]
Has this been tried on the Dev site?

**Comment by harkenn** [comment]
No, I spoke with cmfcmf beforehand. It absolutely needs some facetime on dev. Not only because we need to bullet-proof the utf8 -> utf8mb4 changeover and check for data loss, but we also need to be monitoring all the adventure data for data loss because of the entity migration. @cmfcmf Its definitely premature to jump straight into a ticket like this, for others who may not be familiar with the potential pitfalls, you can't let enthusiasm get too much hold on you.

**Comment by JohnnyFlash** [comment]
SQL dump is in the dev server root home directory.  Let's make some magic happen :)

**Comment by cmfcmf** [comment]
> Has this been tried on the Dev site?
> SQL dump is in the dev server root home directory. Let's make some magic happen :)

Thanks! I've executed the update on the dev site. I only had to adjust a few more things in the data migration command to accomondate for a few new adventures. The dev server is now running on the `adjustments-to-data-migration` branch.

> No, I spoke with cmfcmf beforehand. It absolutely needs some facetime on dev. Not only because we need to bullet-proof the utf8 -> utf8mb4 changeover and check for data loss, but we also need to be monitoring all the adventure data for data loss because of the entity migration. 

If you have an idea on how to do that, feel free to share 👍 

> @cmfcmf Its definitely premature to jump straight into a ticket like this, for others who may not be familiar with the potential pitfalls, you can't let enthusiasm get too much hold on you because the presence of the ticket implies a readiness for it to occur.

Well, I didn't want to run you over. Maybe my wording was a bit off. I intended to say 
> ***Once we decide to do the upgrade***, @JohnnyFlash ***should*** temporarily shutdown the site, create another database dump for us to test, and if it works, execute the upgrade and re-enable the site

To clarify, I'm less driven by enthusiasm and more by fear, the fear being that we make changes and continue developing forever but never update the real production site. I created this ticket to track things that need to be done before the update can - finally - happen.

**Comment by harkenn** [comment]
@JohnnyFlash did you take the site offline? new.adventurelookup.com/adventures is down atm

**Comment by JohnnyFlash** [comment]
So how's it going?  The dev site looks updated... is it done?

**Comment by harkenn** [comment]
@cmfcmf have you updated it?

**Comment by cmfcmf** [comment]
I had updated the dev site in https://github.com/AdventureLookup/AdventureLookup/issues/89#issuecomment-319313057 already. I just now tested the update again using the actual dev which includes the changes of #91. I have not touched anything related to utf8mb4.

**Comment by MattColville** [comment]
It sounds like this merge, if successful, is the last big change before Making A Video. Is that correct?

**Comment by harkenn** [comment]
I don't think this will include the "linking to bad sites" or the new visual design. Otherwise, yes

**Comment by JohnnyFlash** [comment]
@cmfcmf Can you review the steps listed above and make the PR to master?

**Comment by cmfcmf** [comment]
Sure, I'll do so in about 30 minutes.

Jerod Bennett <notifications@github.com> schrieb am Do., 3. Aug. 2017,
20:08:

**Comment by cmfcmf** [comment]
> @cmfcmf Can you review the steps listed above and make the PR to master?

Done in https://github.com/AdventureLookup/AdventureLookup/pull/92. The comments on that PR are all outdated comments given on some of the commits and can be ignored. The steps [at the top](https://github.com/AdventureLookup/AdventureLookup/issues/89#issue-246894493) of this issue are accurate and were executed on the dev site without an issue.

**Comment by harkenn** [comment]
@cmfcmf just happened to get home and be able to review. Good to merge!

**Comment by harkenn** [comment]
done!

**Comment by MattColville** [comment]
I'm not worried about the Whitelisting issue, I'm confident we can implement that after the launch and until then I can manually scrub links out of entries.

The visual stuff I'm not worried about. I looks fine for now and when the new visual look goes live it will look even better!

@JohnnyFlash Can you let me know when the real site goes live?

**Comment by JohnnyFlash** [comment]
Site has been updated!
